resin/models/device.py

```diff
@@ -686,34 +688,42 @@ def disable_device_url(self, uuid):
             endpoint=self.settings.get('pine_endpoint')
         )
 
-    def set_to_build(self, uuid, build):
+    def set_to_build(self, uuid, build_commit_hash):
         """
-        Set a device to specific build id.
+        Set a device to specific build commit hash.
 
         Args:
             uuid (str): device uuid.
-            build (str): build id.
+            build_commit_hash (str): build commit hash.
 
         Raises:
             DeviceNotFound: if device couldn't be found.
             ApplicationNotFound: if application couldn't be found.
             IncompatibleApplication: if moving a device to an application with different device-type.
 
         Examples:
-            >> > resin.models.device.set_to_build('8deb12a58e3b6d3920db1c2b6303d1ff32f23d5ab99781ce1dde6876e8d143', '123098')
+            >> > resin.models.device.set_to_build('4457d5db93be458270666ef8b8157c66', '950eac2d3efce555490c96e7c9b55c37b385acb6')
             'OK'
 
         """
 
         if not self.has(uuid):
             raise exceptions.DeviceNotFound(uuid)
 
+        app_id = self.get(uuid)['belongs_to__application']['__id']
+        build_id = None
+        for build in self.build.get_all_by_application(app_id):
```

Comment:
This can bring in an unbounded number of builds, which can be a few. You can fetch a specific build by its commit hash directly with an OData request.

Reply:
I see, updated. Thanks @dfunckt